File: OtherLanguages/PHP/CrashReporter/Data/EcsRequest.php
<?
class EcsRequest
{
	private $action;
	private $data;

	public function read($reader)
	{
		$this->action = $reader->readString(0);
        
        if($reader->hasField(1))
        {
            $this->data = $reader->readStructure(1);
        }
	}
    
    public function getAction()
    {
        return $this->action;
    }
    
    public function getData()
    {
        return $this->data;
    }
}
?>

File: Sitana.Framework/DataTransfer/Ecs/EcsRequest.cs
using System;

namespace Ebatianos.DataTransfer
{
	public class EcsRequest: IEcsStructure
	{
        public object StateObject;

		public string Action;
		public IEcsStructure Data;

        internal EcsHttpClient.ResponseHandler Handler;

        void IEcsStructure.Read(EcsReader reader)
		{
            throw new NotImplementedException("This structure is only to send.");
		}

        void IEcsStructure.Write(EcsWriter writer)
		{
			writer.Write(0, Action);
			writer.Write(1, Data);
		}
	}
}